libsolutil/InvertibleMap.h

```diff
@@ -32,26 +32,31 @@ struct InvertibleMap
 
 	void set(K _key, V _value)
 	{
-		if (values.count(_key))
-			references[values[_key]].erase(_key);
-		values[_key] = _value;
+		auto it = values.find(_key);
+		if (it != values.end())
+			references[it->second].erase(_key);
+		it->second = _value;
```

Comment:
The last line might write to ``values.end()`` now, won't it? I think this should something like
```
auto [it, didInsert] = values.insert(std::make_pair(_key, _value));
if (didInsert) references[it->second].erase(_key);
else it->second = _value;
```
(haven't made sure this will be a pair of rvalue references like that and work in general - but *something* like that)

Reply:
oh right!

Reply:
But actually I think we can't blindly insert, because we need the old value.